.github/workflows/ci.yml

```diff
@@ -187,3 +187,28 @@ jobs:
 
     - name: Check formatting
       run: cargo fmt -- --check
+
+  rustsemver:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@a40b8845c0683271d9f53dfcb887a7e181d3918b # 0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - uses: actions/checkout@v3
+
+      - name: Install rust nightly
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
```

Comment:
Do we need `protoc` for building the docs?

Reply:
Yup, otherwise we would get 

```
Error: Failed when running cargo-doc on....

failed to run custom build command for `libp2p-core v0.36.0 (/home/runner/work/rust-libp2p/rust-libp2p/core)`

Caused by:
  process didn't exit successfully: `/home/runner/work/rust-libp2p/rust-libp2p/target/semver-checks/target/debug/build/libp2p-core-ed4cd93bb4d7b68e/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at '
  Could not find `protoc` installation and this build crate cannot proceed without
  this knowledge.
```